src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -389,7 +390,11 @@ def _make_link_value(value: XMLValue, iri_resolver: IriResolver) -> dict[str, An
     elif resolved_iri := iri_resolver.get(s):
         iri = resolved_iri
     else:
-        raise BaseError(f"Could not resolve ID {s} to IRI.")
+        msg = (
+            f"Could not resolve ID {s} to IRI. This is probably because the resource '{s}' could not be created. "
```

Comment:
Or maybe you could say, could not find the IRI in the id2iri mapping file?

Reply:
For me, that would be too much from a insider perspective. I guess that anyone who doesn't know the code of DSP-TOOLS very well does not know what a "id2iri mapping" is.

Reply:
Well they know the file, it is saved in their cwd and they know to keep it.

Reply:
I would also write "could not find ID in the id2iri mapping" , it makes it more clear what the error is about than writing "could not resolve ID to IRI". And maybe also from a general point of view:  it could be better to write exactly what the problem is than trying to write easier messages for 'outsiders' if we want to spread knowledge about code and practices in DaSCH as much as possible.

Reply:
That's true, they know the file name. I have adapted the error message.